File: autotrader/ui.py
import tkinter as tk
from tkinter import ttk
from .bot import create_client, moving_average_cross


def run_ui():
    root = tk.Tk()
    root.title("Binance Trading Bot")

    tk.Label(root, text="API Key").grid(row=0, column=0, sticky="e")
    api_key_entry = tk.Entry(root, width=40)
    api_key_entry.grid(row=0, column=1)

    tk.Label(root, text="API Secret").grid(row=1, column=0, sticky="e")
    api_secret_entry = tk.Entry(root, width=40, show="*")
    api_secret_entry.grid(row=1, column=1)

    tk.Label(root, text="Symbol").grid(row=2, column=0, sticky="e")
    symbol_entry = tk.Entry(root)
    symbol_entry.insert(0, "BTCUSDT")
    symbol_entry.grid(row=2, column=1)

    tk.Label(root, text="Short MA").grid(row=3, column=0, sticky="e")
    ma_short_entry = tk.Entry(root)
    ma_short_entry.insert(0, "5")
    ma_short_entry.grid(row=3, column=1)

    tk.Label(root, text="Long MA").grid(row=4, column=0, sticky="e")
    ma_long_entry = tk.Entry(root)
    ma_long_entry.insert(0, "20")
    ma_long_entry.grid(row=4, column=1)

    tk.Label(root, text="Quantity").grid(row=5, column=0, sticky="e")
    quantity_entry = tk.Entry(root)
    quantity_entry.insert(0, "0.001")
    quantity_entry.grid(row=5, column=1)

    result_var = tk.StringVar()

    def start():
        api_key = api_key_entry.get().strip()
        api_secret = api_secret_entry.get().strip()
        symbol = symbol_entry.get().strip()
        ma_short = int(ma_short_entry.get())
        ma_long = int(ma_long_entry.get())
        quantity = float(quantity_entry.get())

        client = create_client(api_key, api_secret)
        result = moving_average_cross(
            client,
            symbol=symbol,
            ma_short=ma_short,
            ma_long=ma_long,
            quantity=quantity,
        )
        result_var.set(str(result))

    ttk.Button(root, text="Start Trading", command=start).grid(row=6, column=0, columnspan=2, pady=10)
    ttk.Label(root, textvariable=result_var).grid(row=7, column=0, columnspan=2)

    root.mainloop()


if __name__ == "__main__":
    run_ui()

File: ui.php
<?php
// Simple web UI for entering Binance credentials and strategy options
if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    $apiKey = $_POST['api_key'] ?? '';
    $secret = $_POST['api_secret'] ?? '';
    $symbol = $_POST['symbol'] ?? 'BTCUSDT';
    $maShort = (int)($_POST['ma_short'] ?? 5);
    $maLong = (int)($_POST['ma_long'] ?? 20);
    $quantity = (float)($_POST['quantity'] ?? 0.001);

    require __DIR__ . '/trader.php';
    try {
        $result = moving_average_cross($apiKey, $secret, $symbol, $maShort, $maLong, $quantity);
        $message = var_export($result, true);
    } catch (Throwable $e) {
        $message = 'Error: ' . $e->getMessage();
    }
} else {
    $message = '';
}
?>
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Binance Trading Bot</title>
    <style>
        body { font-family: Arial, sans-serif; margin: 2em; }
        label { display: block; margin-top: 0.5em; }
    </style>
</head>
<body>
    <h1>Binance Trading Bot</h1>
    <form method="post">
        <label>API Key
            <input type="text" name="api_key" size="40" required>
        </label>
        <label>API Secret
            <input type="password" name="api_secret" size="40" required>
        </label>
        <label>Symbol
            <input type="text" name="symbol" value="BTCUSDT">
        </label>
        <label>Short MA
            <input type="number" name="ma_short" value="5">
        </label>
        <label>Long MA
            <input type="number" name="ma_long" value="20">
        </label>
        <label>Quantity
            <input type="text" name="quantity" value="0.001">
        </label>
        <button type="submit">Start Trading</button>
    </form>
    <pre><?= htmlspecialchars($message) ?></pre>
</body>
</html>
